lib/rules/compat-engines.ts

```diff
@@ -213,23 +213,21 @@ export default createRule("compat-engines", {
       const depEngines = getEngines(meta);
       for (const module of ctx.engines) {
         const selfVer = selfEngines.get(module)!;
-        const engineValue = depEngines.get(module);
-        if (engineValue) {
-          ctx.markAsProcessed(module);
-          const depVer = getSemverRange(engineValue);
-
-          if (depVer) {
-            if (
-              semver.subset(
-                selfVer.adjust,
-                buildAdjustRangeForDeps(comparisonType, depVer),
-              )
-            ) {
-              ctx.markAsValid(module);
-            } else {
-              ctx.addInvalid(module, depVer);
-            }
-          }
+        const engineValue = depEngines.get(module) || "*";
+        ctx.markAsProcessed(module);
+        const depVer = getSemverRange(engineValue);
```

Comment:
I think we probably need to write it like this:

```suggestion
        const engineValue = depEngines.get(module);
        if (engineValue)
          ctx.markAsProcessed(module);
        const depVer = getSemverRange(engineValue || "*");
```

Reply:
The purpose of this change is to avoid calling `markAsProcessed()` if no `engines` are specified.
This is because if the process is complete for all versions, deep dependencies are not checked.

Reply:
Let me give it a try.